src/modules/75-cf/pages/feature-flags/components/FeatureFlagsListing.tsx

```diff
@@ -63,18 +64,38 @@ const FeatureFlagsListing: FC<FeatureFlagsListingProps> = ({
 }) => {
   const gitSync = useFFGitSyncContext()
   const { getString } = useStrings()
-
+  const [checkedFlags, setCheckedFlags] = useState<Feature[]>([])
+  const [selectAll, setSelectAll] = useState(false)
   const [flagFilter] = useQueryParamsState<Optional<FilterProps>>('filter', {})
 
-  const onCheckboxSelect = useCallback((event: FormEvent<HTMLInputElement>) => {
-    if (event.currentTarget.checked) {
-      //TODO: Handle checked
+  const { FFM_8344_FLAG_CLEANUP } = useFeatureFlags()
+
+  const isStaleFlagsView =
+    FFM_8344_FLAG_CLEANUP && flagFilter?.queryProps?.value === FeatureFlagStatus.POTENTIALLY_STALE
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handleCheckboxChange = (e: any, rowData: any): void => {
+    const { checked } = e.target
+    if (checked) {
+      const tempChecked = [...checkedFlags]
+      tempChecked.push(rowData.value.identifier)
+      setCheckedFlags(tempChecked)
     } else {
-      //TODO: Handle unchecked
+      const filterTemp = checkedFlags.filter(uncheckRow => uncheckRow !== rowData.value.identifier)
+      setCheckedFlags(filterTemp)
     }
-  }, [])
+  }
 
-  const { FFM_8344_FLAG_CLEANUP } = useFeatureFlags()
+  const handleSelectAll = useCallback(() => {
+    setSelectAll(!selectAll)
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    setCheckedFlags(features?.features?.map((feat: any) => feat.identifier).filter(y => y !== undefined) ?? [])
```

Comment:
_(NP)_ it might make more sense to filter out the undefined items before mapping the output
```suggestion
    setCheckedFlags(features?.features?.filter((feat: Feature) => !!feat.identifier)?.map((feat: Feature) => feat.identifier) ?? [])
```